sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoinExec.scala

```diff
@@ -454,6 +490,43 @@ case class BroadcastHashJoinExec(
     val (matched, checkCondition, _) = getJoinCondition(ctx, input)
     val numOutput = metricTerm(ctx, "numOutputRows")
 
+    if (isNullAwareAntiJoin) {
+      if (broadcastRelation.value.isInstanceOf[EmptyHashedRelation]) {
+        return s"""
+                  |// NAAJ Join EmptyHashedRelation accept all
```

Comment:
`NAAJ` already means Join. How about `If the right side is empty, NAAJ simply returns the left side.`

Reply:
done